dbms/src/DataStreams/HashJoinProbeBlockInputStream.cpp

```diff
@@ -127,6 +127,7 @@ Block HashJoinProbeBlockInputStream::getOutputBlock()
         {
         case ProbeStatus::PROBE:
         {
+            join->waitUntilAllBuildFinished();
```

Comment:
Suggest to add a new state for waitingForAllBuildFinished.

Reply:
This new state is only useful for BlockInputStream, not for the pipeline model, because the build/probe relationship is guaranteed by the scheduler under the pipeline model.
So I personally give praise :)